pkg/virt-controller/watch/clone/clone.go

```diff
@@ -259,6 +262,18 @@ func (ctrl *VMCloneController) syncSourceVMTargetVM(key string, source *k6tv1.Vi
 		ctrl.logAndRecord(vmClone, TargetVMCreated, fmt.Sprintf("created target VM %s for clone %s", targetVMInfo.Name, vmClone.Name))
 		syncInfo.targetVMCreated = true
 
+		targetVM, ok := targetVMObj.(*k6tv1.VirtualMachine)
+		if !ok {
+			return addErrorToSyncInfo(syncInfo, fmt.Errorf("VM object from vm informer cannot be converted to *VirtualMachine: %v", err))
+		}
+
+		if targetVM.Status.RestoreInProgress != nil {
+			return addErrorToSyncInfo(syncInfo, fmt.Errorf("status of restore %s still not updated in VM %s status", *targetVM.Status.RestoreInProgress, targetVM.Name))
```

Comment:
Do I read this right that we are reporting a real error to the controller loop? I am not sure this is an error, that seems to be like a normal we-have-to-wait-until-it-is-done check. In this case we normally don't return errors (printing a log message for debug purposes is of course fine). We would even continue updating the status section if not in sync, but we would not return with an error, which would lead to a back-off retry. Not 100% sure this is what is happening here ,but it looks like it.

Reply:
Yes, you're reading it correctly.

The reason I chose this path is since I didn't want to pass a VM informer and add an event handler to it. With the current implementation, the clone controller does not track VMs. My concern was that it would introduce an overhead to track all VMs, wake up for every one of their changes, check if the change is relevant (99% it's not), then go back to sleep.

It's important to note that this is now being added since there's a small delay between the restore reflecting that it's ready and the VM reflecting that the restore is ready. This delay is expected to be very small, so I thought returning an error and trying again is reasonable, perhaps more reasonable than waking up for every VM change with a very very low probability of the change being relevant.

p.s. in the original clone implementation, the `SyncInfo` struct had information on requeuing. Perhaps I can do the same now - simply requeue (after, say, 1 second) in this scenario. This would require some more changes, which again I'm not sure are worth the trouble for this corner case, but it's an option.

@mhenriks @rmohr WDYT?

Reply:
Probably best to wait for @mhenriks. I was not part of the design decisions on this one, but I would be cautious. This really reports an error, like including it in our controller-prometheus metrics, making it hard to see real errors. And if I understand you correct, we just talk about an eventually-consistency-race which is pretty normal, and not a "faulty" race, right?

Reply:
Yes, you got it right. Thanks for raising this concern.
Alright, let's wait for @mhenriks to bring his thoughts

Reply:
My preference would be that the clone layer does not have to know/care about `targetVM.Status.RestoreInProgress`.  I think it would be better if the restore is not complete until `targetVM.Status.RestoreInProgress` is nil.  But that will take a bit of work in the restore controller.

For now, maybe the best path is to simply have the test wait for `RestoreInProgress` to be nil before starting the VM?  Then we can refactor the restore controller at our convenience?

Reply:
> My preference would be that the clone layer does not have to know/care about `targetVM.Status.RestoreInProgress`. I think it would be better if the restore is not complete until `targetVM.Status.RestoreInProgress` is nil. But that will take a bit of work in the restore controller.
> 
> For now, maybe the best path is to simply have the test wait for `RestoreInProgress` to be nil before starting the VM? Then we can refactor the restore controller at our convenience?

Sounds great :+1: 
Let's try that.